diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/EmptyBlock.kt

```diff
@@ -82,18 +91,33 @@ class EmptyBlock(configRules: List<RulesConfig>) : DiktatRule(
         }
     }
 
-    @Suppress("UnsafeCallOnNullableType", "WRONG_WHITESPACE")
-    private fun isAnonymousSamClass(node: ASTNode) : Boolean =
+    @Suppress("UnsafeCallOnNullableType")
+    private fun isAnonymousSamClass(node: ASTNode): Boolean =
             if (node.elementType == FUNCTION_LITERAL && node.hasParent(CALL_EXPRESSION)) {
-                // We are checking identifier because it is not class in AST
-                // , SAM conversions are indistinguishable from lambdas.
+                // We are checking identifier because it is not class in AST,
+                // SAM conversions are indistinguishable from lambdas.
                 // So we just verify that identifier is in PascalCase
                 val valueArgument = node.findParentNodeWithSpecificType(CALL_EXPRESSION)!!
                 valueArgument.findLeafWithSpecificType(IDENTIFIER)?.text?.isPascalCase() ?: false
             } else {
                 false
             }
 
+    @Suppress("UnsafeCallOnNullableType")
+    private fun isLambdaUsedAsFunction(node: ASTNode): Boolean {
+        val parents = node.parents()
+        if (parents.any { it.elementType == CALL_EXPRESSION }) {
+            val callExpression = parents.find { it.elementType == CALL_EXPRESSION }!!
+            // excepting cases like list.map { }
+            return callExpression.treeParent.elementType != DOT_QUALIFIED_EXPRESSION
+        } else if (parents.any { it.elementType == LAMBDA_EXPRESSION }) {
+            val lambdaExpression = parents.find { it.elementType == LAMBDA_EXPRESSION }!!
+            // cases like A({}). Here Lambda expression is used as a value parameter.
```

Comment:
I remember that we have e.g. this case: https://github.com/cqfn/diKTat/blob/d9b6b6eff3f189aca9ac15f6ba25d230f047ff0c/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/KotlinParser.kt#L53, where the compiler suggests to remove explicit class name as redundant, but diktat would've raised an error. Please try to remove and see if you PR fixes it.

Reply:
Fixed